src/autopas/LogicHandler.h

```diff
@@ -560,16 +557,56 @@ class LogicHandler {
    * If the locks are already initialized but the container size changed, surplus locks will
    * be deleted, new locks are allocated and locks that are still necessary are reused.
    *
-   * @param boxLength
-   * @param interactionLengthInv
+   * @note Should the generated number of locks exceed 1e6, the number of locks is reduced so that
+   * the number of locks per dimensions is proportional to the domain side lengths and smaller than the limit.
+   *
+   * @param boxLength Size per dimension for the box that should be covered by locks (should include halo).
+   * @param interactionLengthInv Inverse of the side length of the virtual boxes one lock is responsible for.
+   *
    */
   void initSpacialLocks(const std::array<double, 3> &boxLength, double interactionLengthInv) {
     using namespace autopas::utils::ArrayMath::literals;
     using autopas::utils::ArrayMath::ceil;
     using autopas::utils::ArrayUtils::static_cast_copy_array;
 
-    // one lock per interaction length + one for each halo region
-    const auto locksPerDim = static_cast_copy_array<size_t>(ceil(boxLength * interactionLengthInv) + 2.);
+    // The maximum number of spatial locks is capped at 1e6.
+    // Without this cap, very large domains (or tiny cutoffs) would generate an insane number of locks,
+    // that could blow up the memory.
+    constexpr size_t maxNumSpacialLocks{1000000};
```

Comment:
I think this limitation should be documented at the member variable as well, probably together with a reference to the function:

```cpp
  /**
   * Locks for regions in the domain. Used for buffer <-> container interaction.
   * <something like capped to 1e6 in initialization function xy>
   */
  std::vector<std::vector<std::vector<std::unique_ptr<std::mutex>>>> _spacialLocks;
```

If I understand correctly, `1e6` is an educated magic number. If so, it should also probably be stated somewhere.

Reply:
Mentioned limit at declaration of `_spacialLocks` -> 5e7347d
"Justified" the choice of `1e6` -> 4818bd2